lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -294,62 +242,44 @@ if Enum.all?(
       {{:ok, actions}, state}
     end
 
-    if Enum.all?(@opus_deps, &Code.ensure_loaded?/1) do
-      defp hls_links_and_children(link_builder, :OPUS, track, _framerate) do
-        %ParentSpec{
-          children: %{
-            {:track_receiver, track.id} => %TrackReceiver{
-              track: track,
-              initial_target_variant: :high
-            },
-            {:depayloader, track.id} => Track.get_depayloader(track),
-            {:opus_decoder, track.id} => Membrane.Opus.Decoder,
-            {:aac_encoder, track.id} => Membrane.AAC.FDK.Encoder,
-            {:aac_parser, track.id} => %Membrane.AAC.Parser{out_encapsulation: :none}
+    defp hls_links_and_children(link_builder, %Track{encoding: :OPUS} = track, _state) do
+      %ParentSpec{
+        children: %{
+          {:track_receiver, track.id} => %TrackReceiver{
+            track: track,
+            initial_target_variant: :high
           },
-          links: [
-            link_builder
-            |> to({:track_receiver, track.id})
-            |> to({:depayloader, track.id})
-            |> to({:opus_decoder, track.id})
-            |> to({:aac_encoder, track.id})
-            |> to({:aac_parser, track.id})
-            |> via_in(Pad.ref(:input, {:audio, track.id}), options: [encoding: :AAC])
-            |> to({:hls_sink_bin, track.stream_id})
-          ]
-        }
-      end
-    else
-      defp hls_links_and_children(_link_builder, :OPUS, _track, _framerate) do
-        raise """
-        Cannot find one of the modules required to support Opus audio input.
-        Ensure `:membrane_opus_plugin`, `:membrane_aac_plugin` and `:membrane_aac_fdk_plugin` are added to the deps.
-        """
-      end
```

Comment:
With that removed in the current version and with missing deps, the module will be simply unavailable. That might be confusing, so it would be perfect to bring back some kind of error in case of missing deps. BTW it wasn't perfect in the first place as with missing H246 same thing was happening.

I'm thinking of moving the top level `if` inside of the module and adding `else` that would define a raising `handle_init` but I'm open to any other ideas. And of course, this may be done in a separate PR